modules/convert/convert.go

```diff
@@ -30,28 +30,76 @@ func ToEmail(email *models.EmailAddress) *api.Email {
 }
 
 // ToBranch convert a git.Commit and git.Branch to an api.Branch
-func ToBranch(repo *models.Repository, b *git.Branch, c *git.Commit, bp *models.ProtectedBranch, user *models.User) *api.Branch {
+func ToBranch(repo *models.Repository, b *git.Branch, c *git.Commit, bp *models.ProtectedBranch, user *models.User, isRepoAdmin bool) *api.Branch {
 	if bp == nil {
 		return &api.Branch{
-			Name:                b.Name,
-			Commit:              ToCommit(repo, c),
-			Protected:           false,
-			RequiredApprovals:   0,
-			EnableStatusCheck:   false,
-			StatusCheckContexts: []string{},
-			UserCanPush:         true,
-			UserCanMerge:        true,
+			Name:                          b.Name,
+			Commit:                        ToCommit(repo, c),
+			Protected:                     false,
+			RequiredApprovals:             0,
+			EnableStatusCheck:             false,
+			StatusCheckContexts:           []string{},
+			UserCanPush:                   true,
+			UserCanMerge:                  true,
+			EffectiveBranchProtectionName: "",
+			EffectiveBRanchProtectionID:   0,
 		}
 	}
+	branchProtectionName := ""
+	var branchProtectionID int64
+	if isRepoAdmin {
+		branchProtectionName = bp.BranchName
+		branchProtectionID = bp.ID
+	}
+
 	return &api.Branch{
-		Name:                b.Name,
-		Commit:              ToCommit(repo, c),
-		Protected:           true,
-		RequiredApprovals:   bp.RequiredApprovals,
-		EnableStatusCheck:   bp.EnableStatusCheck,
-		StatusCheckContexts: bp.StatusCheckContexts,
-		UserCanPush:         bp.CanUserPush(user.ID),
-		UserCanMerge:        bp.CanUserMerge(user.ID),
+		Name:                          b.Name,
+		Commit:                        ToCommit(repo, c),
+		Protected:                     true,
+		RequiredApprovals:             bp.RequiredApprovals,
+		EnableStatusCheck:             bp.EnableStatusCheck,
+		StatusCheckContexts:           bp.StatusCheckContexts,
+		UserCanPush:                   bp.CanUserPush(user.ID),
+		UserCanMerge:                  bp.CanUserMerge(user.ID),
+		EffectiveBranchProtectionName: branchProtectionName,
+		EffectiveBRanchProtectionID:   branchProtectionID,
```

Comment:
I'm not sure the user needs to know the database ID. I guess it's for retrieving details later. However, I think those details could and should be requested by name when possible.

Reply:
Yes to retrieve the details as I use the ID in the endpoint path. An alternative would be to change the endpoint paths to
`​/repos​/{owner}​/{repo}​/branch_protections​/{branch}`
where `{branch}` is either an actual branch or later an wildcard branch protection. It still separates branches from branch protection. The drawbacks:
 - Wildcard protected branches will have to be url encoded in the path when requested
 - You will not be able to change what branch(es) the protection covers. This is maybe not a big deal.

Reply:
I'd say we get rid of the IDs. 😁